pkg/reconciler/knativeserving/ingress/istio.go

```diff
@@ -18,23 +18,36 @@ package ingress
 
 import (
 	"context"
+	"strconv"
 
 	mf "github.com/manifestival/manifestival"
 	"go.uber.org/zap"
 	istionetworkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
 	"istio.io/client-go/pkg/clientset/versioned/scheme"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"knative.dev/operator/pkg/apis/operator/base"
 	"knative.dev/operator/pkg/apis/operator/v1beta1"
 	servingv1beta1 "knative.dev/operator/pkg/apis/operator/v1beta1"
 	"knative.dev/pkg/logging"
 )
 
+const enableSecretInformerFilteringByCertUIDAnno = "knative.dev/enable-secret-informer-filtering-by-cert-uid"
+
 var istioFilter = ingressFilter("istio")
 
+func init() {
+	_ = appsv1.AddToScheme(scheme.Scheme)
```

Comment:
Needed for the deployment kind.

Reply:
I still don't think we need this init() function. In the transformer, when the `enable-secret-informer-filtering-by-cert-uid` is enabled, the deployment `net-istio-controller` will be changed and returned.
The list of resources in the manifest is changed, and applied in the reconciling loop, so it should be good for Kubernetes to pick up the changes.